sql: TestTxnAutoRetryParallelStmts failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/8bd6caec6eed07ecff29801f3f11993352544387

Parameters:
```
TAGS=
GOFLAGS=-race
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=845746&tab=buildLog

```
=== RUN   TestTxnAutoRetryParallelStmts
W180818 06:03:14.643732 168180 server/status/runtime.go:294  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180818 06:03:14.854694 168180 server/server.go:829  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I180818 06:03:14.855366 168180 base/addr_validation.go:260  [n?] server certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180818 06:03:14.855447 168180 base/addr_validation.go:300  [n?] web UI certificate addresses: IP=127.0.0.1,::1; DNS=localhost,*.local; CN=node
I180818 06:03:14.925473 168180 server/config.go:496  [n?] 1 storage engine initialized
I180818 06:03:14.925745 168180 server/config.go:499  [n?] RocksDB cache size: 128 MiB
I180818 06:03:14.925818 168180 server/config.go:499  [n?] store 0: in-memory, size 0 B
I180818 06:03:15.179873 168180 server/node.go:373  [n?] **** cluster d342b1eb-ffa9-46f6-a178-367a9bd343b9 has been created
I180818 06:03:15.180119 168180 server/server.go:1400  [n?] **** add additional nodes by specifying --join=127.0.0.1:40055
I180818 06:03:15.181523 168180 gossip/gossip.go:383  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:40055" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:0 patch:0 unstable:11 > build_tag:"v2.1.0-alpha.20180702-1745-g8bd6cae" started_at:1534572195181048833 
I180818 06:03:15.271134 168180 storage/store.go:1509  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180818 06:03:15.272235 168180 server/node.go:476  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=6.9 KiB), ranges=1, leases=1, queries=0.00, writes=0.00, bytesPerReplica={p10=7103.00 p25=7103.00 p50=7103.00 p75=7103.00 p90=7103.00 pMax=7103.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I180818 06:03:15.272839 168180 storage/stores.go:242  [n1] read 0 node addresses from persistent storage
I180818 06:03:15.273630 168180 server/node.go:697  [n1] connecting to gossip network to verify cluster ID...
I180818 06:03:15.273995 168180 server/node.go:722  [n1] node connected via gossip and verified as part of cluster "d342b1eb-ffa9-46f6-a178-367a9bd343b9"
I180818 06:03:15.274489 168180 server/node.go:546  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180818 06:03:15.321099 168180 server/status/recorder.go:652  [n1] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I180818 06:03:15.321182 168180 server/server.go:1806  [n1] Could not start heap profiler worker due to: directory to store profiles could not be determined
I180818 06:03:15.331007 168470 server/status/recorder.go:652  [n1,summaries] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I180818 06:03:15.335568 168180 server/server.go:1537  [n1] starting https server at 127.0.0.1:32981 (use: 127.0.0.1:32981)
I180818 06:03:15.338698 168180 server/server.go:1539  [n1] starting grpc/postgres server at 127.0.0.1:40055
I180818 06:03:15.343052 168180 server/server.go:1540  [n1] advertising CockroachDB node at 127.0.0.1:40055
W180818 06:03:15.343462 168180 jobs/registry.go:308  [n1] unable to get node liveness: node not in the liveness table
I180818 06:03:15.361544 168349 storage/replica_command.go:282  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I180818 06:03:15.682208 168230 storage/replica_command.go:282  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
I180818 06:03:16.168770 168482 sql/event_log.go:126  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I180818 06:03:16.420968 168486 storage/replica_command.go:282  [split,n1,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I180818 06:03:16.740342 168353 rpc/nodedialer/nodedialer.go:92  [consistencyChecker,n1,s1,r1/1:/{Min-System/}] connection to n1 established
I180818 06:03:16.775423 168241 storage/replica_command.go:282  [split,n1,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
I180818 06:03:17.149073 168006 storage/replica_command.go:282  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
I180818 06:03:17.514591 168443 sql/event_log.go:126  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:$1 User:root}
I180818 06:03:17.559919 168530 storage/replica_command.go:282  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I180818 06:03:18.057606 168490 sql/event_log.go:126  [n1,intExec=disableNetTrace] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:root}
I180818 06:03:18.104072 168127 storage/replica_command.go:282  [split,n1,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I180818 06:03:18.267115 168580 storage/replica_command.go:282  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
W180818 06:03:18.372508 168554 storage/intent_resolver.go:668  [n1,s1] failed to push during intent resolution: failed to push "split" id=88ea4820 key=/Local/Range/Table/11/RangeDescriptor rw=true pri=0.00080564 iso=SERIALIZABLE stat=PENDING epo=0 ts=1534572198.267354454,0 orig=1534572198.267354454,0 max=1534572198.267354454,0 wto=false rop=false seq=1
I180818 06:03:18.429676 168581 storage/replica_command.go:282  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180818 06:03:18.736107 168596 storage/replica_command.go:282  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180818 06:03:18.839202 168612 sql/event_log.go:126  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:gen_random_uuid()::STRING User:root}
I180818 06:03:19.103047 168627 storage/replica_command.go:282  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180818 06:03:19.239802 168572 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I180818 06:03:19.467761 168606 storage/replica_command.go:282  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180818 06:03:19.587898 168017 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I180818 06:03:19.739431 168653 storage/replica_command.go:282  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180818 06:03:19.753633 168180 server/server.go:1593  [n1] done ensuring all necessary migrations have run
I180818 06:03:19.757884 168180 server/server.go:1596  [n1] serving sql connections
I180818 06:03:20.004464 168654 server/server_update.go:67  [n1] no need to upgrade, cluster already at the newest version
I180818 06:03:20.008030 168625 sql/event_log.go:126  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:40055} Attrs: Locality: ServerVersion:2.0-11 BuildTag:v2.1.0-alpha.20180702-1745-g8bd6cae StartedAt:1534572195181048833 LocalityAddress:[]} ClusterID:d342b1eb-ffa9-46f6-a178-367a9bd343b9 StartedAt:1534572195181048833 LastUp:1534572195181048833}
I180818 06:03:20.025757 168723 storage/replica_command.go:282  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I180818 06:03:20.206730 168683 storage/replica_command.go:282  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I180818 06:03:20.564145 168747 storage/replica_command.go:282  [split,n1,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I180818 06:03:20.816491 168770 storage/replica_command.go:282  [split,n1,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I180818 06:03:21.052643 168694 storage/replica_command.go:282  [split,n1,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180818 06:03:21.243760 168785 storage/replica_command.go:282  [split,n1,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
I180818 06:03:21.536145 168672 storage/replica_command.go:282  [split,n1,s1,r20/1:/{Table/23-Max}] initiating a split of this range at key /Table/50 [r21]
I180818 06:03:21.771297 168792 storage/replica_command.go:282  [split,n1,s1,r21/1:/{Table/50-Max}] initiating a split of this range at key /Table/51 [r22]
I180818 06:03:22.752966 168866 storage/replica_command.go:282  [split,n1,s1,r22/1:/{Table/51-Max}] initiating a split of this range at key /Table/52 [r23]
I180818 06:03:22.769029 168850 sql/event_log.go:126  [n1,client=127.0.0.1:51310,user=root] Event: "create_database", target: 52, info: {DatabaseName:t Statement:CREATE DATABASE t User:root}
I180818 06:03:22.969248 168850 sql/event_log.go:126  [n1,client=127.0.0.1:51310,user=root] Event: "create_table", target: 53, info: {TableName:t.public.test Statement:CREATE TABLE t.public.test (k INT PRIMARY KEY, v TEXT, t DECIMAL) User:root}
I180818 06:03:23.040052 168801 storage/replica_command.go:282  [split,n1,s1,r23/1:/{Table/52-Max}] initiating a split of this range at key /Table/53 [r24]
I180818 06:03:23.239644 168768 storage/replica_command.go:282  [split,n1,s1,r24/1:/{Table/53-Max}] initiating a split of this range at key /Table/54 [r25]
I180818 06:03:23.240901 168850 sql/event_log.go:126  [n1,client=127.0.0.1:51310,user=root] Event: "create_table", target: 54, info: {TableName:t.public.test2 Statement:CREATE TABLE t.public.test2 (k INT PRIMARY KEY, v TEXT, t DECIMAL) User:root}
I180818 06:03:23.565860 168709 storage/replica_command.go:282  [split,n1,s1,r25/1:/{Table/54-Max}] initiating a split of this range at key /Table/55 [r26]
I180818 06:03:23.568649 168850 sql/event_log.go:126  [n1,client=127.0.0.1:51310,user=root] Event: "create_table", target: 55, info: {TableName:t.public.test3 Statement:CREATE TABLE t.public.test3 (k INT PRIMARY KEY, v TEXT, t DECIMAL) User:root}
--- FAIL: TestTxnAutoRetryParallelStmts (14.79s)
=== RUN   TestTxnAutoRetryParallelStmts/retryStmt1=true,retryStmt2=true,retryStmt3=true
I180818 06:03:28.079418 169440 storage/replica_command.go:72  [n1,s1,r24/1:/Table/5{3-4}] test injecting error: TransactionAbortedError: txn aborted "sql txn" id=b785b732 key=/Table/53/1/1/0 rw=true pri=0.00642589 iso=SERIALIZABLE stat=PENDING epo=0 ts=1534572208.072022621,0 orig=1534572208.072022621,0 max=1534572208.072022621,0 wto=false rop=false seq=2
==================
WARNING: DATA RACE
Read at 0x00c420813020 by goroutine 1433:
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).OrigTimestamp()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:267 +0x43
  github.com/cockroachdb/cockroach/pkg/sql.(*TableCollection).getTableVersion()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:271 +0x774
  github.com/cockroachdb/cockroach/pkg/sql.(*CachedPhysicalAccessor).GetObjectDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/physical_schema_accessors.go:201 +0x1b6
  github.com/cockroachdb/cockroach/pkg/sql.(*LogicalSchemaAccessor).GetObjectDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/logical_schema_accessors.go:79 +0x3c6
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).LookupObject()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/resolver.go:207 +0x27f
  github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*TableName).ResolveExisting()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/name_resolution.go:310 +0x32b
  github.com/cockroachdb/cockroach/pkg/sql.ResolveExistingObject()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/resolver.go:92 +0x162
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).Insert()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/insert.go:84 +0x19c
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).newPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:743 +0x3b7f
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).makePlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:303 +0x175
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInParallel()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:635 +0x10e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:387 +0xd3f
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x46e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1079 +0x3413
  github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:416 +0x1e0
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func3()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:267 +0x1a9

Previous write at 0x00c420813020 by goroutine 632:
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).replaceSenderIfTxnAbortedLocked()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:899 +0x66b
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:798 +0x528
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Send-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:490 +0xa3
  github.com/cockroachdb/cockroach/pkg/internal/client.sendAndFill()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:542 +0x151
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:490 +0xfd
  github.com/cockroachdb/cockroach/pkg/sql.(*tableWriterBase).finalize()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/tablewriter.go:162 +0x176
  github.com/cockroachdb/cockroach/pkg/sql.(*tableInserter).finalize()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/tablewriter_insert.go:57 +0x75
  github.com/cockroachdb/cockroach/pkg/sql.(*insertNode).BatchedNext()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/insert.go:457 +0x3ad
  github.com/cockroachdb/cockroach/pkg/sql.(*rowCountNode).startExec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_batch.go:173 +0x143
  github.com/cockroachdb/cockroach/pkg/sql.startExec.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:568 +0xd3
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:127 +0x99
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:552 +0x583
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:94 +0xfb
  github.com/cockroachdb/cockroach/pkg/sql.walkPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:58 +0x1c1
  github.com/cockroachdb/cockroach/pkg/sql.startExec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:573 +0x152
  github.com/cockroachdb/cockroach/pkg/sql.startPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:505 +0x74
  github.com/cockroachdb/cockroach/pkg/sql.(*planTop).start()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:478 +0x10d
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithLocalEngine()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:887 +0x193
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInParallel.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:677 +0x526
  github.com/cockroachdb/cockroach/pkg/sql.(*ParallelizeQueue).Add.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/parallel_stmts.go:118 +0x14f

Goroutine 1433 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:266 +0x1444
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.serveConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:162 +0x240
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:453 +0x9b2
  github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func19.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1618 +0x17e
  github.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:141 +0xd9

Goroutine 632 (finished) created at:
  github.com/cockroachdb/cockroach/pkg/sql.(*ParallelizeQueue).Add()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/parallel_stmts.go:89 +0x1e1
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInParallel()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:658 +0x4bd
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:387 +0xd3f
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x46e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1079 +0x3413
  github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:416 +0x1e0
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func3()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:267 +0x1a9
==================
==================
WARNING: DATA RACE
Write at 0x00c420bd0c00 by goroutine 1433:
  sync/atomic.CompareAndSwapInt32()
      /usr/local/go/src/runtime/race_amd64.s:293 +0xb
  sync.(*Mutex).Lock()
      /usr/local/go/src/sync/mutex.go:74 +0x4d
  github.com/cockroachdb/cockroach/pkg/util/syncutil.(*Mutex).Lock()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/syncutil/mutex_sync.go:32 +0x3a
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).OrigTimestamp()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:947 +0x4f
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).OrigTimestamp()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:267 +0x60
  github.com/cockroachdb/cockroach/pkg/sql.(*TableCollection).getTableVersion()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:271 +0x774
  github.com/cockroachdb/cockroach/pkg/sql.(*CachedPhysicalAccessor).GetObjectDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/physical_schema_accessors.go:201 +0x1b6
  github.com/cockroachdb/cockroach/pkg/sql.(*LogicalSchemaAccessor).GetObjectDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/logical_schema_accessors.go:79 +0x3c6
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).LookupObject()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/resolver.go:207 +0x27f
  github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*TableName).ResolveExisting()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/name_resolution.go:310 +0x32b
  github.com/cockroachdb/cockroach/pkg/sql.ResolveExistingObject()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/resolver.go:92 +0x162
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).Insert()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/insert.go:84 +0x19c
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).newPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:743 +0x3b7f
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).makePlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:303 +0x175
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInParallel()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:635 +0x10e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:387 +0xd3f
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x46e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1079 +0x3413
  github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:416 +0x1e0
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func3()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:267 +0x1a9

Previous write at 0x00c420bd0c00 by goroutine 632:
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSenderFactory).TransactionalSender()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:404 +0xf0
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).replaceSenderIfTxnAbortedLocked()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:899 +0x633
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:798 +0x528
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Send-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:490 +0xa3
  github.com/cockroachdb/cockroach/pkg/internal/client.sendAndFill()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:542 +0x151
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:490 +0xfd
  github.com/cockroachdb/cockroach/pkg/sql.(*tableWriterBase).finalize()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/tablewriter.go:162 +0x176
  github.com/cockroachdb/cockroach/pkg/sql.(*tableInserter).finalize()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/tablewriter_insert.go:57 +0x75
  github.com/cockroachdb/cockroach/pkg/sql.(*insertNode).BatchedNext()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/insert.go:457 +0x3ad
  github.com/cockroachdb/cockroach/pkg/sql.(*rowCountNode).startExec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_batch.go:173 +0x143
  github.com/cockroachdb/cockroach/pkg/sql.startExec.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:568 +0xd3
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:127 +0x99
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:552 +0x583
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:94 +0xfb
  github.com/cockroachdb/cockroach/pkg/sql.walkPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:58 +0x1c1
  github.com/cockroachdb/cockroach/pkg/sql.startExec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:573 +0x152
  github.com/cockroachdb/cockroach/pkg/sql.startPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:505 +0x74
  github.com/cockroachdb/cockroach/pkg/sql.(*planTop).start()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:478 +0x10d
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithLocalEngine()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:887 +0x193
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInParallel.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:677 +0x526
  github.com/cockroachdb/cockroach/pkg/sql.(*ParallelizeQueue).Add.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/parallel_stmts.go:118 +0x14f

Goroutine 1433 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:266 +0x1444
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.serveConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:162 +0x240
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:453 +0x9b2
  github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func19.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1618 +0x17e
  github.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:141 +0xd9

Goroutine 632 (finished) created at:
  github.com/cockroachdb/cockroach/pkg/sql.(*ParallelizeQueue).Add()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/parallel_stmts.go:89 +0x1e1
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInParallel()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:658 +0x4bd
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:387 +0xd3f
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x46e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1079 +0x3413
  github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:416 +0x1e0
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func3()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:267 +0x1a9
==================
==================
WARNING: DATA RACE
Write at 0x00c420bd0c08 by goroutine 1433:
  sync/atomic.StoreInt32()
      /usr/local/go/src/runtime/race_amd64.s:229 +0xb
  github.com/cockroachdb/cockroach/pkg/util/syncutil.(*Mutex).Lock()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/syncutil/mutex_sync.go:33 +0x54
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).OrigTimestamp()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:947 +0x4f
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).OrigTimestamp()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:267 +0x60
  github.com/cockroachdb/cockroach/pkg/sql.(*TableCollection).getTableVersion()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:271 +0x774
  github.com/cockroachdb/cockroach/pkg/sql.(*CachedPhysicalAccessor).GetObjectDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/physical_schema_accessors.go:201 +0x1b6
  github.com/cockroachdb/cockroach/pkg/sql.(*LogicalSchemaAccessor).GetObjectDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/logical_schema_accessors.go:79 +0x3c6
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).LookupObject()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/resolver.go:207 +0x27f
  github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*TableName).ResolveExisting()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/name_resolution.go:310 +0x32b
  github.com/cockroachdb/cockroach/pkg/sql.ResolveExistingObject()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/resolver.go:92 +0x162
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).Insert()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/insert.go:84 +0x19c
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).newPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:743 +0x3b7f
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).makePlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:303 +0x175
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInParallel()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:635 +0x10e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:387 +0xd3f
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x46e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1079 +0x3413
  github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:416 +0x1e0
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func3()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:267 +0x1a9

Previous write at 0x00c420bd0c08 by goroutine 632:
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSenderFactory).TransactionalSender()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:404 +0xf0
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).replaceSenderIfTxnAbortedLocked()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:899 +0x633
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:798 +0x528
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Send-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:490 +0xa3
  github.com/cockroachdb/cockroach/pkg/internal/client.sendAndFill()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:542 +0x151
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:490 +0xfd
  github.com/cockroachdb/cockroach/pkg/sql.(*tableWriterBase).finalize()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/tablewriter.go:162 +0x176
  github.com/cockroachdb/cockroach/pkg/sql.(*tableInserter).finalize()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/tablewriter_insert.go:57 +0x75
  github.com/cockroachdb/cockroach/pkg/sql.(*insertNode).BatchedNext()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/insert.go:457 +0x3ad
  github.com/cockroachdb/cockroach/pkg/sql.(*rowCountNode).startExec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_batch.go:173 +0x143
  github.com/cockroachdb/cockroach/pkg/sql.startExec.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:568 +0xd3
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:127 +0x99
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:552 +0x583
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:94 +0xfb
  github.com/cockroachdb/cockroach/pkg/sql.walkPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:58 +0x1c1
  github.com/cockroachdb/cockroach/pkg/sql.startExec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:573 +0x152
  github.com/cockroachdb/cockroach/pkg/sql.startPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:505 +0x74
  github.com/cockroachdb/cockroach/pkg/sql.(*planTop).start()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:478 +0x10d
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithLocalEngine()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:887 +0x193
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInParallel.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:677 +0x526
  github.com/cockroachdb/cockroach/pkg/sql.(*ParallelizeQueue).Add.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/parallel_stmts.go:118 +0x14f

Goroutine 1433 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:266 +0x1444
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.serveConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:162 +0x240
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:453 +0x9b2
  github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func19.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1618 +0x17e
  github.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:141 +0xd9

Goroutine 632 (finished) created at:
  github.com/cockroachdb/cockroach/pkg/sql.(*ParallelizeQueue).Add()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/parallel_stmts.go:89 +0x1e1
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInParallel()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:658 +0x4bd
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:387 +0xd3f
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x46e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1079 +0x3413
  github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:416 +0x1e0
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func3()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:267 +0x1a9
==================
==================
WARNING: DATA RACE
Read at 0x00c420bd0ca0 by goroutine 1433:
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).OrigTimestamp()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:949 +0x93
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).OrigTimestamp()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:267 +0x60
  github.com/cockroachdb/cockroach/pkg/sql.(*TableCollection).getTableVersion()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:271 +0x774
  github.com/cockroachdb/cockroach/pkg/sql.(*CachedPhysicalAccessor).GetObjectDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/physical_schema_accessors.go:201 +0x1b6
  github.com/cockroachdb/cockroach/pkg/sql.(*LogicalSchemaAccessor).GetObjectDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/logical_schema_accessors.go:79 +0x3c6
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).LookupObject()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/resolver.go:207 +0x27f
  github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*TableName).ResolveExisting()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/name_resolution.go:310 +0x32b
  github.com/cockroachdb/cockroach/pkg/sql.ResolveExistingObject()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/resolver.go:92 +0x162
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).Insert()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/insert.go:84 +0x19c
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).newPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:743 +0x3b7f
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).makePlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:303 +0x175
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInParallel()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:635 +0x10e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:387 +0xd3f
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x46e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1079 +0x3413
  github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:416 +0x1e0
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func3()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:267 +0x1a9

Previous write at 0x00c420bd0ca0 by goroutine 632:
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSenderFactory).TransactionalSender()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:404 +0xf0
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).replaceSenderIfTxnAbortedLocked()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:899 +0x633
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:798 +0x528
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Send-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:490 +0xa3
  github.com/cockroachdb/cockroach/pkg/internal/client.sendAndFill()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:542 +0x151
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:490 +0xfd
  github.com/cockroachdb/cockroach/pkg/sql.(*tableWriterBase).finalize()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/tablewriter.go:162 +0x176
  github.com/cockroachdb/cockroach/pkg/sql.(*tableInserter).finalize()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/tablewriter_insert.go:57 +0x75
  github.com/cockroachdb/cockroach/pkg/sql.(*insertNode).BatchedNext()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/insert.go:457 +0x3ad
  github.com/cockroachdb/cockroach/pkg/sql.(*rowCountNode).startExec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_batch.go:173 +0x143
  github.com/cockroachdb/cockroach/pkg/sql.startExec.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:568 +0xd3
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:127 +0x99
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:552 +0x583
  github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:94 +0xfb
  github.com/cockroachdb/cockroach/pkg/sql.walkPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:58 +0x1c1
  github.com/cockroachdb/cockroach/pkg/sql.startExec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:573 +0x152
  github.com/cockroachdb/cockroach/pkg/sql.startPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:505 +0x74
  github.com/cockroachdb/cockroach/pkg/sql.(*planTop).start()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:478 +0x10d
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithLocalEngine()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:887 +0x193
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInParallel.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:677 +0x526
  github.com/cockroachdb/cockroach/pkg/sql.(*ParallelizeQueue).Add.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/parallel_stmts.go:118 +0x14f

Goroutine 1433 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:266 +0x1444
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.serveConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:162 +0x240
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/server.go:453 +0x9b2
  github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func19.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1618 +0x17e
  github.com/cockroachdb/cockroach/pkg/util/netutil.(*Server).ServeWith.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:141 +0xd9

Goroutine 632 (finished) created at:
  github.com/cockroachdb/cockroach/pkg/sql.(*ParallelizeQueue).Add()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/parallel_stmts.go:89 +0x1e1
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInParallel()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:658 +0x4bd
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:387 +0xd3f
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:95 +0x46e
  github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1079 +0x3413
  github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:416 +0x1e0
  github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func3()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:267 +0x1a9
==================
I180818 06:03:28.106792 169523 storage/replica_command.go:72  [n1,s1,r24/1:/Table/5{3-4}] test injecting error: TransactionRetryError: retry txn (RETRY_POSSIBLE_REPLAY): "sql txn" id=2093c921 key=/Table/53/1/1/0 rw=true pri=0.01638113 iso=SERIALIZABLE stat=PENDING epo=1 ts=1534572208.080654245,0 orig=1534572208.080654245,0 max=1534572208.102200372,0 wto=false rop=false seq=2
I180818 06:03:28.125352 169528 storage/replica_command.go:72  [n1,s1,r26/1:/{Table/55-Max}] test injecting error: TransactionAbortedError: txn aborted "sql txn" id=2093c921 key=/Table/53/1/1/0 rw=true pri=0.04232819 iso=SERIALIZABLE stat=PENDING epo=3 ts=1534572208.080654245,0 orig=1534572208.080654245,0 max=1534572208.102200372,0 wto=false rop=false seq=3
I180818 06:03:28.214182 169499 storage/replica_command.go:72  [n1,s1,r24/1:/Table/5{3-4}] test injecting error: TransactionAbortedError: txn aborted "sql txn" id=bf9f5340 key=/Table/53/1/1/0 rw=true pri=0.04232819 iso=SERIALIZABLE stat=PENDING epo=1 ts=1534572208.126766292,0 orig=1534572208.126766292,0 max=1534572208.194232924,0 wto=false rop=false seq=2
I180818 06:03:28.284220 169502 storage/replica_command.go:72  [n1,s1,r24/1:/Table/5{3-4}] test injecting error: TransactionRetryError: retry txn (RETRY_POSSIBLE_REPLAY): "sql txn" id=83ccc6be key=/Table/55/1/3/0 rw=true pri=0.04232819 iso=SERIALIZABLE stat=PENDING epo=1 ts=1534572208.215599607,0 orig=1534572208.215599607,0 max=1534572208.265215193,0 wto=false rop=false seq=3
I180818 06:03:28.284971 169503 storage/replica_command.go:72  [n1,s1,r25/1:/Table/5{4-5}] test injecting error: TransactionRetryError: retry txn (RETRY_POSSIBLE_REPLAY): "sql txn" id=83ccc6be key=/Table/55/1/3/0 rw=true pri=0.04232819 iso=SERIALIZABLE stat=PENDING epo=1 ts=1534572208.215599607,0 orig=1534572208.215599607,0 max=1534572208.266667724,0 wto=false rop=false seq=4
I180818 06:03:28.285682 169504 storage/replica_command.go:72  [n1,s1,r26/1:/{Table/55-Max}] test injecting error: TransactionAbortedError: txn aborted "sql txn" id=83ccc6be key=/Table/55/1/3/0 rw=true pri=0.04232819 iso=SERIALIZABLE stat=PENDING epo=1 ts=1534572208.215599607,0 orig=1534572208.215599607,0 max=1534572208.265005113,0 wto=false rop=false seq=2
I180818 06:03:28.348893 169514 storage/replica_command.go:72  [n1,s1,r26/1:/{Table/55-Max}] test injecting error: TransactionRetryError: retry txn (RETRY_POSSIBLE_REPLAY): "sql txn" id=d773623d key=/Table/55/1/3/0 rw=true pri=0.05208040 iso=SERIALIZABLE stat=PENDING epo=1 ts=1534572208.302876394,0 orig=1534572208.302876394,0 max=1534572208.327724995,0 wto=false rop=false seq=2
I180818 06:03:28.365791 169513 storage/replica_command.go:72  [n1,s1,r25/1:/Table/5{4-5}] test injecting error: TransactionRetryError: retry txn (RETRY_POSSIBLE_REPLAY): "sql txn" id=d773623d key=/Table/55/1/3/0 rw=true pri=0.05208040 iso=SERIALIZABLE stat=PENDING epo=1 ts=1534572208.302876394,0 orig=1534572208.302876394,0 max=1534572208.342892405,0 wto=false rop=false seq=3
I180818 06:03:28.376549 169544 storage/replica_command.go:72  [n1,s1,r25/1:/Table/5{4-5}] test injecting error: TransactionAbortedError: txn aborted "sql txn" id=d773623d key=/Table/55/1/3/0 rw=true pri=0.05208040 iso=SERIALIZABLE stat=PENDING epo=3 ts=1534572208.302876394,0 orig=1534572208.302876394,0 max=1534572208.327724995,0 wto=false rop=false seq=3
I180818 06:03:28.433454 168144 storage/replica_proposal.go:214  [n1,s1,r5/1:/System/ts{d-e}] new range lease repl=(n1,s1):1 seq=3 start=1534572195.213687173,0 epo=1 pro=1534572208.428019954,0 following repl=(n1,s1):1 seq=2 start=1534572195.213687173,0 exp=1534572204.242203808,0 pro=1534572195.242298581,0
I180818 06:03:28.446401 168424 storage/replica_proposal.go:214  [n1,s1,r21/1:/Table/5{0-1}] new range lease repl=(n1,s1):1 seq=3 start=1534572195.213687173,0 epo=1 pro=1534572208.378664197,0 following repl=(n1,s1):1 seq=2 start=1534572195.213687173,0 exp=1534572204.242203808,0 pro=1534572195.242298581,0
I180818 06:03:28.481223 169586 storage/replica_command.go:72  [n1,s1,r25/1:/Table/5{4-5}] test injecting error: TransactionAbortedError: txn aborted "sql txn" id=05a250e1 key=/Table/53/1/1/0 rw=true pri=0.05208040 iso=SERIALIZABLE stat=PENDING epo=1 ts=1534572208.413498021,0 orig=1534572208.413498021,0 max=1534572208.462125339,0 wto=false rop=false seq=4
I180818 06:03:28.576354 169555 storage/replica_command.go:72  [n1,s1,r26/1:/{Table/55-Max}] test injecting error: TransactionRetryError: retry txn (RETRY_POSSIBLE_REPLAY): "sql txn" id=05a250e1 key=/Table/53/1/1/0 rw=true pri=0.05208040 iso=SERIALIZABLE stat=PENDING epo=1 ts=1534572208.413498021,0 orig=1534572208.413498021,0 max=1534572208.460525026,0 wto=false rop=false seq=3
I180818 06:03:28.773015 168180 util/stop/stopper.go:537  quiescing; tasks left:
3      node.Node: batch
1      ts.poller: poll
1      storage.intentResolver: processing intents
1      kv.DistSender: sending partial batch
1      [async] transport racer
I180818 06:03:28.789616 168180 util/stop/stopper.go:537  quiescing; tasks left:
2      node.Node: batch
1      ts.poller: poll
1      storage.intentResolver: processing intents
1      [async] transport racer
I180818 06:03:28.795879 168180 util/stop/stopper.go:537  quiescing; tasks left:
1      ts.poller: poll
1      node.Node: batch
1      [async] transport racer
I180818 06:03:28.967500 168452 kv/transport_race.go:67  transport race promotion: ran 129 iterations on up to 1445 requests
I180818 06:03:28.968173 168180 util/stop/stopper.go:537  quiescing; tasks left:
1      ts.poller: poll
1      node.Node: batch
I180818 06:03:29.152953 168180 util/stop/stopper.go:537  quiescing; tasks left:
1      ts.poller: poll
W180818 06:03:29.153533 168469 ts/db.go:195  [n1,ts-poll] error writing time series data: result is ambiguous (server shutdown)
    --- FAIL: TestTxnAutoRetryParallelStmts/retryStmt1=true,retryStmt2=true,retryStmt3=true (0.80s)
    	txn_restart_test.go:183: checking error injection failed
    	testing.go:730: race detected during execution of test
	testing.go:730: race detected during execution of test

```

--- a/pkg/internal/client/txn.go
+++ b/pkg/internal/client/txn.go
@@ -264,6 +264,12 @@ func (txn *Txn) Isolation() enginepb.IsolationType {
 // committing so this is not guaranteed to be the commit timestamp.
 // Use CommitTimestamp() when needed.
 func (txn *Txn) OrigTimestamp() hlc.Timestamp {
+	txn.mu.Lock()
+	defer txn.mu.Unlock()
+	return txn.origTimestampLocked()
+}
+
+func (txn *Txn) origTimestampLocked() hlc.Timestamp {
 	return txn.mu.sender.OrigTimestamp()
 }
 
@@ -272,18 +278,24 @@ func (txn *Txn) OrigTimestamp() hlc.Timestamp {
 // method will guarantee that if a timestamp push is needed
 // the commit will fail with a retryable error.
 func (txn *Txn) CommitTimestamp() hlc.Timestamp {
+	txn.mu.Lock()
+	defer txn.mu.Unlock()
 	return txn.mu.sender.CommitTimestamp()
 }
 
 // CommitTimestampFixed returns true if the commit timestamp has
 // been fixed to the start timestamp and cannot be pushed forward.
 func (txn *Txn) CommitTimestampFixed() bool {
+	txn.mu.Lock()
+	defer txn.mu.Unlock()
 	return txn.mu.sender.CommitTimestampFixed()
 }
 
 // SetSystemConfigTrigger sets the system db trigger to true on this transaction.
 // This will impact the EndTransactionRequest.
 func (txn *Txn) SetSystemConfigTrigger() error {
+	txn.mu.Lock()
+	defer txn.mu.Unlock()
 	txn.systemConfigTrigger = true
 	return txn.mu.sender.SetSystemConfigTrigger()
 }
@@ -296,6 +308,8 @@ func (txn *Txn) SetSystemConfigTrigger() error {
 // DisablePipelining must be called before any operations are performed on the
 // transaction.
 func (txn *Txn) DisablePipelining() error {
+	txn.mu.Lock()
+	defer txn.mu.Unlock()
 	return txn.mu.sender.DisablePipelining()
 }
 
@@ -559,7 +573,7 @@ func (txn *Txn) UpdateDeadlineMaybe(ctx context.Context, deadline hlc.Timestamp)
 	txn.mu.Lock()
 	defer txn.mu.Unlock()
 	if txn.mu.deadline == nil || deadline.Less(*txn.mu.deadline) {
-		if deadline.Less(txn.OrigTimestamp()) {
+		if deadline.Less(txn.origTimestampLocked()) {
 			log.Fatalf(ctx, "deadline below txn.OrigTimestamp() is nonsensical; "+
 				"txn has would have no change to commit. Deadline: %s", deadline)
 		}

